libs/orders/src/lib/components/order-list/order-edit-dialog.tsx

```diff
@@ -37,15 +37,15 @@ export const OrderEditDialog = ({
     register,
     formState: { errors },
     handleSubmit,
-  } = useForm<FormFields>({
-    defaultValues: {
-      entryPrice: order?.price
-        ? addDecimal(order?.price, order?.market?.decimalPlaces ?? 0)
-        : '',
-    },
-  });
+    setValue,
+  } = useForm<FormFields>();
 
   if (!order) return null;
+  const step = toDecimal(order.market?.decimalPlaces ?? 0);
+  setValue(
+    'entryPrice',
+    addDecimalsFormatNumber(order.price, order.market?.decimalPlaces ?? 0)
+  );
```

Comment:
chore (blocking?): sorry for holding this up again, i think it would feel a lot less complicated to render this dialog from the parent with a required order object, and leave the defaultValues as it was, so you don't need to return null from the component, and don't need to forcefully reset the field value on every render.

```
{order && < OrderEditDialog  order={order} />}
```

Reply:
Yup in React if you are setting state in render you are 100% doing something wrong.

Reply:
yeah, my bad. updated.